src/pages/Message.jsx

```diff
@@ -1,134 +1,299 @@
+      }
+
+      const msgs = [];
+      subcollectionSnapshott.forEach((subDoc) => {
+        const letter = subDoc.data();
+        msgs.push({
+          collectionId: subDoc.id,
+          attachments: letter.attachments,
+          letter: letter.letter,
+          sent_by: letter.sent_by,
+          status: letter.status,
+          created_at: letter.created_at,
+          moderation: letter.moderation_comments,
+        });
+        console.log('letter', letter)
+        setLastMessageDoc(subDoc);
+      });
+      if (user) {
+        console.log('user found in get sub data')
+        const pendingQ = query(
+          subcollectionRe,
+          where("status", "==", 'pending_review'),
+          where("deleted_at", '==', null),
+          orderBy("created_at", "desc"),
+          where("sent_by", "==", user),
+          limit(PAGE_SIZE)
+        );
+        const pendingSubcollectionSnapshott = await getDocs(pendingQ);
+        if (!pendingSubcollectionSnapshott.empty) {
+          pendingSubcollectionSnapshott.forEach((subDoc) => {
+            const letter = subDoc.data();
+            msgs.push({
+              collectionId: subDoc.id,
+              attachments: letter.attachments,
+              letter: letter.letter,
+              sent_by: letter.sent_by,
+              status: letter.status,
+              created_at: letter.created_at,
+              moderation: letter.moderation_comments,
+              pending: true
+            });
+          })
         }
-      } catch (error) {
-        console.error('Error fetching user:', error);
       }
-    };
+      setMessages(msgs)
+      setMessageDocRef(documentRe)
 
-    fetchData();
-  }, []);
+      const draftQuery = query(
+        subcollectionRe,
+        where("status", "==", "draft"),
+        where("deleted_at", "==", null),
+        where("sent_by", "==", user),
+        orderBy("created_at", "desc"),
+        limit(1)
+      );
 
-  useEffect(() => {
-    console.log("new id", childId)
-    let isMounted = true;
-
-    if (!childId) return
-    const fetchData = async () => {
-      try {
-        const collectionRef = collection(firestore, "Chat");
-        const chatBuddy = window.location.pathname.split("/messages/")[1];
-        const intlBuddyRef = doc(collection(firestore, 'InternationalBuddy'), chatBuddy);
-        const childReference = doc(collection(firestore, 'Child'), childId);
-        setChildRef(childReference);
-
-        const q = query(
-          collectionRef,
-          where("child", '==', childReference),
-          where("international_buddy", '==', intlBuddyRef)
-        );
+      const draftSnapshot = await getDocs(draftQuery);
+      if (!draftSnapshot.empty) {
+        const draftData = draftSnapshot.docs[0].data();
+        setDraft({
+          id: draftSnapshot.docs[0].id,
+          attachments: draftData.attachments,
+          letter: draftData.letter
+        });
+      } else {
+        setDraft(null);
+      }
```

Comment:
I certainly don't know why without this `if (user)` block we run into two console errors, but we do. I noticed you [use it](https://github.com/murphy-charitable-foundation/magic-chat-app-web/pull/4/files#diff-c8852b18d7a0b8f4b688392de7ab3082af77bb486e6ed33ccd2807e23ef0f915R112) for the `pending_review` query, so I suppose you know why; without it, I get the following two errors:
<img width="836" alt="Screenshot 2024-03-11 at 10 37 55 PM" src="https://github.com/murphy-charitable-foundation/magic-chat-app-web/assets/31979020/21314337-ac6e-4500-8ba1-1666e9074718">
```suggestion
      if (user) {
        const draftQuery = query(
          subcollectionRe,
          where("status", "==", "draft"),
          where("deleted_at", "==", null),
          where("sent_by", "==", user),
          orderBy("created_at", "desc"),
          limit(1)
        );

        const draftSnapshot = await getDocs(draftQuery);
        if (!draftSnapshot.empty) {
          const draftData = draftSnapshot.docs[0].data();
          setDraft({
            id: draftSnapshot.docs[0].id,
            attachments: draftData.attachments,
            letter: draftData.letter
          });
        } else {
          setDraft(null);
        }
      }
```

Reply:
the reason is on page load we are trying to fetch the data but the firestore user has not been found yet, so by wrapping it in the if we can make sure we don't make fetches we know will break.